js/ui/panel.js

```diff
@@ -2921,33 +2921,26 @@ Panel.prototype = {
         if (iconSize === -1) { // Legacy: Scale to panel size
             iconSize = height;
         } else if (iconSize === 0) { // To best fit within the panel size
+            let fitFactor = isSymbolic ? .5 : .8;
+            iconSize = toStandardIconSize(height * fitFactor);
+        } else if (iconSize > height) {
+            // Don't try to fit an icon that is larger than the panel
             iconSize = toStandardIconSize(height);
         }
 
-        // Don't try to fit an icon that is larger than the panel
-        let i = -1;
-        while (iconSize > height) {
-            i++;
-            iconSize = toStandardIconSize(height - i);
-        }
-
-        if (isSymbolic) {
-            iconSize = iconSize * 0.9;
-        }
-
         return iconSize; // Always return a value above 0 or St will spam the log.
     },
 
     getPanelZoneIconSize: function(locationLabel, iconType) {
         let zoneConfig = this._panelZoneIconSizes;
+        let symbolic = iconType === St.IconType.SYMBOLIC;
 
         if (!zoneConfig) {
-            let height = this.height / global.ui_scale;
             global.logError(`[Panel ${this.panelId}] Unable to find zone configuration`);
-            return toStandardIconSize(height);
+            return _calculatePanelZoneIconSize(0, symbolic);
```

Comment:
Hi @germanfr, thanks for the fixes. It looks like `this` is missing here.

Reply:
Oops